File: Cpp/Day05-Loops.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

// https://www.hackerrank.com/contests/30-days-of-code/challenges/day-5-loops/submissions/code/4670907

void series(int a, int b, int N) {
    for (int n = 0; n < N; ++n) {
        
        int bSum = 0;
        
        for (int i = 0; i <= n; ++i)
            bSum += (lround(pow(2, i)) * b);
        
        cout << a + bSum;
        if (n+1 != N)
            cout << " ";
    }
    cout << endl;
}

int main() {
    int t;
    cin >> t;
    
    while (t --> 0) {
        int a, b, n;
        cin >> a >> b >> n;
        series(a, b, n);
    }
    
    return 0;
}

File: Java8/Day05-Loops.java
import java.io.*;
import java.util.*;

// https://www.hackerrank.com/contests/30-days-of-code/challenges/day-5-loops/submissions/code/4682102

public class Solution {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        for (int cases = scn.nextInt(); cases > 0; --cases) {
            
            int a = scn.nextInt();
            int b = scn.nextInt();
            int n = scn.nextInt();
            
            int sum = 0;
            for (int i = 0; i < n; i++) {
                sum += (int)(Math.pow(2, i) * b);
                if (i != 0)
                    System.out.print(" ");
                System.out.print(sum+a);
            }
            if (cases > 1)
                System.out.println();
        }
    }
}
